pointcloud/kdtree.go

```diff
@@ -268,21 +268,32 @@ func (kd *KDTree) KNearestNeighbors(p r3.Vector, k int, includeSelf bool) []*Poi
 // as the first element with distance 0.
 func (kd *KDTree) RadiusNearestNeighbors(p r3.Vector, r float64, includeSelf bool) []*PointAndData {
 	keep := kdtree.NewDistKeeper(r)
-	kd.tree.NearestSet(keep, &treeComparableR3Vector{p})
+	kd.Tree.NearestSet(keep, &treeComparableR3Vector{p})
 	return keeperToArray(keep.Heap, kd.points, p, includeSelf, math.MaxInt)
 }
 
 // Iterate iterates over all points in the cloud.
 func (kd *KDTree) Iterate(numBatches, myBatch int, fn func(p r3.Vector, d Data) bool) {
-	kd.tree.Do(func(c kdtree.Comparable, b *kdtree.Bounding, depth int) bool {
+	kd.Tree.Do(func(c kdtree.Comparable, b *kdtree.Bounding, depth int) bool {
 		p, ok := c.(treeComparableR3Vector)
 		if !ok {
 			panic("Comparable is not a Point")
 		}
 		d, ok := kd.points.At(p.vec.X, p.vec.Y, p.vec.Z)
 		if !ok {
-			panic("Mismatch between tree and point storage.")
+			panic("Mismatch between Tree and point storage.")
 		}
 		return !fn(p.vec, d)
 	})
 }
+
+// Iterate iterates over all points in the cloud.
```

Comment:
Had to create this function because of the way I'm preemptively splitting up the tree and trying to run the clustering algorithm on the children at level 2.